### PR TITLE
Revise new color bar features

### PR DESCRIPTION
**Pull request by forman**
This PR comprises any backend changes required for https://github.com/xcube-dev/xcube-viewer/pull/391

Checklist:

* [x] Add unit tests and/or doctests in docstrings
* [x] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] ~New/modified features documented in `docs/source/*`~
* [x] Changes documented in `CHANGES.md`
* [x] GitHub CI passes
* [ ] AppVeyor CI passes
* [x] Test coverage remains or increases (target 100%)

### CODE REVIEW

File: xcube/util/cmaps.py [hunk 1]
```diff
@@ -444,48 +465,55 @@ def _register_ocm_cmaps(self):
...
+                values.append(key)
+                colors.append(color)
+                if i == n - 1:
+                    # the last key's next boundary is key+1
+                    values.append(key + 1)
+                elif key + 1 != int(cm_items[i + 1][0]):
+                    # insert transparent region from key+1 to next key
+                    values.append(key + 1)
+                    colors.append("#00000000")
+            cmap = matplotlib.colors.ListedColormap(colors, name=cm_base_name)
```
Comment (konstntokas) [inline xcube/util/cmaps.py:490]: This does not work properly. 

See example below. The values at the coast are all above 6, as presented in the continuous colormap on the left. However, the categorical colormap on the right finds values for category 4 and category 2. 

Screenshot from 2024-07-18 09-14-47

I think also here, we will not get around using `matplotlib.colors.BoundaryNorm` (see PR #1019)

**Review by konstntokas — CHANGES_REQUESTED** [review]
I think categorical colormap does not work properly. See example below.

File: xcube/util/cmaps.py [hunk 2]
```diff
@@ -444,48 +465,55 @@ def _register_ocm_cmaps(self):
...
+                    next_value = cm_items[i + 1][0]
+                    stepwise_cm_items.append((value, color))
+                    stepwise_cm_items.append((next_value, color))
+                cm_items = stepwise_cm_items
+            cmap = matplotlib.colors.LinearSegmentedColormap.from_list(
+                cm_base_name, cm_items
             )
+        return cm_name, Colormap(
+            cm_base_name,
+            cm_type="node",
```
[the remark this answers is not in the document]
Comment (forman) [reply xcube/util/cmaps.py:512]: It should be the only option left, logically.

File: xcube/util/cmaps.py (hunk 1 above)
Comment (forman) [reply xcube/util/cmaps.py:490]: if `cm_type == "key"`, I indeed use `matplotlib.colors.BoundaryNorm`.

File: xcube/util/cmaps.py [hunk 3]
```diff
@@ -213,72 +214,92 @@ def __init__(self, name: str, desc: str, cm_names: Optional[list[str]] = None):
...
+        self._colors = values
 
     @property
     def cm_name(self) -> str:
+        """The colormap's name."""
         return self._cm_name
 
+    @property
+    def cm_type(self) -> str:
+        """The colormap type, always one of "node", "bound", "key"."""
```
[the remark this answers is not in the document]
Comment (forman) [reply xcube/util/cmaps.py:249]: Internal names for "Continuous", "Stepwise", "Categorical" color bar types in the Viewer UI. I may rename then in another PR.

**Review by konstntokas — APPROVED** [review]
Works as expected now! Great!

**Pull request merged**